crates/prover/src/rpc/utils.rs

```diff
@@ -0,0 +1,234 @@
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use citrea_common::utils::{
+    check_l2_range_exists, extract_sequencer_commitments, filter_out_proven_commitments,
+};
+use jsonrpsee::core::RpcResult;
+use jsonrpsee::types::error::{INTERNAL_ERROR_CODE, INTERNAL_ERROR_MSG};
+use jsonrpsee::types::ErrorObjectOwned;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use sov_db::ledger_db::ProverLedgerOps;
+use sov_db::schema::types::BatchNumber;
+use sov_modules_api::{SlotData, Zkvm};
+use sov_rollup_interface::da::{BlockHeaderTrait, DaSpec, SequencerCommitment};
+use sov_rollup_interface::services::da::DaService;
+use sov_rollup_interface::zk::{StateTransitionData, ZkvmHost};
+use sov_stf_runner::ProverService;
+use tracing::error;
+
+use super::RpcContext;
+use crate::da_block_handler::{
+    break_sequencer_commitments_into_groups, get_state_transition_data_from_commitments,
+};
+
+pub(super) async fn get_sequencer_commitments_for_proving<C, Da, Ps, Vm, DB, StateRoot, Witness>(
```

Comment:
we should also use this function in the da block handler too imo

or else whenever we change anything regarding the order of processing sequencer commitments etc.

now we have to update these places:
- prover da block handler
- full node da block handler
- the function the batch proof zk guest calls (cant remember the name now)
- *and this function*

Reply:
Done